### PR TITLE
fix: print all empty octets as empty strings

### PR DESCRIPTION
**Pull request by matteo-cristino**
- fix: print all empty octets as empty strings
- test(zencode): add failing tests with empty octets in the trace

### CODE REVIEW

**Comment by matteo-cristino** [comment]
@jaromil I think I will need your help for this part https://github.com/dyne/Zenroom/pull/883/files#diff-2d5670cdc813fe0e61b94eb79c029066d90b15a37ddffc012283322702c3920dR199, not quite sure of how to do it

File: test/zencode/zencode_exec.bats [hunk 1]
```diff
@@ -195,3 +195,44 @@ EOF
 	save_output verified.json
 	assert_output '{"myMessage":"Dear Bob, your name is too short, goodbye - Alice.","output":["Zenroom_certifies_that_signatures_are_all_correct!"]}'
 }
+
+# TODO: how to test failure and error message with zencode_exec?
```
Comment (jaromil) [inline test/zencode/zencode_exec.bats:199]: zencode-exec returns Z->exitcode whose values are defined in zenroom.h
```c
// EXIT CODES
#define ERR_INIT 4
#define ERR_PARSE 3
#define ERR_EXEC 2
#define ERR_GENERIC 1 // EXIT_FAILURE
#define SUCCESS 0 // EXIT_SUCCESS
```

Comment (matteo-cristino) [reply test/zencode/zencode_exec.bats]: At the end I was able to catch the message error! Bats seems not to save the output and error when using a `cat file | run executable`

**Comment by jaromil** [comment]
many thanks for this additional test!
